wikipedia_cleanup/data_filter.py

```diff
@@ -257,6 +263,86 @@ def change_pair_needs_to_be_filtered(
         )
 
 
+class FeatureAdderFilter(AbstractDataFilter):
+    def _filter_for_property(self, changes: List[InfoboxChange]) -> List[InfoboxChange]:
+        change_timestamps = pd.Series((change.value_valid_from for change in changes))
+        change_timestamps = (
+            change_timestamps.dt.normalize().dt.normalize().dt.tz_localize(None)
+        )
+        day_of_year = change_timestamps.dt.dayofyear
+        day_of_month = change_timestamps.dt.day
+        day_of_week = change_timestamps.dt.dayofweek
+        month_of_year = change_timestamps.dt.month
+        quarter_of_year = change_timestamps.dt.quarter
+        is_month_start = change_timestamps.dt.is_month_start
+        is_month_end = change_timestamps.dt.is_month_end
+        is_quarter_start = change_timestamps.dt.is_quarter_start
+        is_quarter_end = change_timestamps.dt.is_quarter_end
+
+        days_since_last_change = (
+            change_timestamps
+            - change_timestamps.shift(+1).fillna(pd.Timestamp("20990101"))
+        ).dt.days
+        days_since_last_change = days_since_last_change.copy()
+        days_since_last_change.loc[days_since_last_change < 0] = 0
+
+        days_since_last_2_changes = (
+            change_timestamps
+            - change_timestamps.shift(+2).fillna(pd.Timestamp("20990101"))
+        ).dt.days
+        days_since_last_2_changes = days_since_last_2_changes.copy()
+        days_since_last_2_changes.loc[(days_since_last_2_changes < 0)] = 0
+
+        days_since_last_3_changes = (
+            change_timestamps
+            - change_timestamps.shift(+3).fillna(pd.Timestamp("20990101"))
+        ).dt.days
+        days_since_last_3_changes = days_since_last_3_changes.copy()
+        days_since_last_3_changes.loc[days_since_last_3_changes < 0] = 0
```

Comment:
This should be done possible without the .loc as you are just indexing via booleans *i think*